plugins/modules/firehose/module.go

```diff
@@ -383,6 +390,53 @@ func (m *Module) Act(r *resource.Resource, action string, params map[string]inte
 	return r.Configs, nil
 }
 
+func Log(ctx context.Context, r *resource.Resource, filter map[string]string) (<-chan module.LogChunk, error) {
+	var cfg config
+	if err := mapstructure.Decode(r.Configs, &cfg); err != nil {
+		return nil, errors.New("unable to parse configs")
+	}
+
+	var releaseConfig *helm.ReleaseConfig
+	if err := mapstructure.Decode(r.Configs[releaseConfigString], &releaseConfig); err != nil {
+		return nil, errors.New("unable to parse configs")
+	}
+
+	podLogs, err := kubelogger.GetStreamingLogs(ctx, releaseConfig.Namespace, releaseConfig.Name)
+	if err != nil {
+		return nil, err
+	}
+	logs := make(chan module.LogChunk)
+	defer close(logs)
```

Comment:
channel should be closed when the writer exits. so this should be inside the `go func(){`